From 92260919d7f1d0fe023251876d4dd7627334348a Mon Sep 17 00:00:00 2001
From: Janmm14 <computerjanimaus@yahoo.de>
Date: Sat, 12 Dec 2015 23:43:30 +0100
Subject: [PATCH 3/7] Optional server list ping logging; Add IPs to the log
 where user names are shown.

Server list pings are only logged if the log_server_list_pings config.yml option is true, defaults to false

The log format is like this: [/***.***.***.***|Janmm14]
If the user name is unknown, its just the ip in the brackets as before.
---
 api/src/main/java/net/md_5/bungee/api/ProxyConfig.java     |  5 +++++
 proxy/src/main/java/net/md_5/bungee/ServerConnector.java   |  2 +-
 .../src/main/java/net/md_5/bungee/conf/Configuration.java  |  5 +++++
 .../java/net/md_5/bungee/connection/DownstreamBridge.java  |  2 +-
 .../java/net/md_5/bungee/connection/InitialHandler.java    | 14 ++++++++++++--
 .../java/net/md_5/bungee/connection/UpstreamBridge.java    |  2 +-
 6 files changed, 25 insertions(+), 5 deletions(-)

diff --git a/api/src/main/java/net/md_5/bungee/api/ProxyConfig.java b/api/src/main/java/net/md_5/bungee/api/ProxyConfig.java
index 61c0d14..f71a4fc 100644
--- a/api/src/main/java/net/md_5/bungee/api/ProxyConfig.java
+++ b/api/src/main/java/net/md_5/bungee/api/ProxyConfig.java
@@ -51,6 +51,11 @@ public interface ProxyConfig
     boolean isLogThrottled();
 
     /**
+     * Whether we log throttled join attempts
+     */
+    boolean isLogServerListPing();
+
+    /**
      * Returns the player max.
      */
     int getPlayerLimit();
diff --git a/proxy/src/main/java/net/md_5/bungee/ServerConnector.java b/proxy/src/main/java/net/md_5/bungee/ServerConnector.java
index 5282033..e2ff05d 100644
--- a/proxy/src/main/java/net/md_5/bungee/ServerConnector.java
+++ b/proxy/src/main/java/net/md_5/bungee/ServerConnector.java
@@ -342,6 +342,6 @@ public class ServerConnector extends PacketHandler
     @Override
     public String toString()
     {
-        return "[" + user.getName() + "] <-> ServerConnector [" + target.getName() + "]";
+        return "[" + user.getAddress() + "|" + user.getName() + "] <-> ServerConnector [" + target.getName() + "]";
     }
 }
diff --git a/proxy/src/main/java/net/md_5/bungee/conf/Configuration.java b/proxy/src/main/java/net/md_5/bungee/conf/Configuration.java
index 3bf31de..1e358b9 100644
--- a/proxy/src/main/java/net/md_5/bungee/conf/Configuration.java
+++ b/proxy/src/main/java/net/md_5/bungee/conf/Configuration.java
@@ -55,6 +55,10 @@ public class Configuration implements ProxyConfig
      * Whether we log throttled join attempts
      */
     private boolean logThrottled;
+    /**
+     * Whether we log server list pings
+     */
+    private boolean logServerListPing;
     private int playerLimit = -1;
     private Collection<String> disabledCommands;
     private int throttle = 4000;
@@ -86,6 +90,7 @@ public class Configuration implements ProxyConfig
         onlineMode = adapter.getBoolean( "online_mode", onlineMode );
         logCommands = adapter.getBoolean( "log_commands", logCommands );
         logThrottled = adapter.getBoolean( "log_throttled", logThrottled );
+        logServerListPing = adapter.getBoolean( "log_server_list_ping", logServerListPing );
         playerLimit = adapter.getInt( "player_limit", playerLimit );
         throttle = adapter.getInt( "connection_throttle", throttle );
         tabCompleteThrottle = adapter.getInt( "tab_complete_throttle", tabCompleteThrottle );
diff --git a/proxy/src/main/java/net/md_5/bungee/connection/DownstreamBridge.java b/proxy/src/main/java/net/md_5/bungee/connection/DownstreamBridge.java
index 96a866e..3ff2f9b 100644
--- a/proxy/src/main/java/net/md_5/bungee/connection/DownstreamBridge.java
+++ b/proxy/src/main/java/net/md_5/bungee/connection/DownstreamBridge.java
@@ -499,6 +499,6 @@ public class DownstreamBridge extends PacketHandler
     @Override
     public String toString()
     {
-        return "[" + con.getName() + "] <-> DownstreamBridge <-> [" + server.getInfo().getName() + "]";
+        return "[" + con.getAddress() + "|" + con.getName() + "] <-> DownstreamBridge <-> [" + server.getInfo().getName() + "]";
     }
 }
diff --git a/proxy/src/main/java/net/md_5/bungee/connection/InitialHandler.java b/proxy/src/main/java/net/md_5/bungee/connection/InitialHandler.java
index d747d9a..601a05f 100644
--- a/proxy/src/main/java/net/md_5/bungee/connection/InitialHandler.java
+++ b/proxy/src/main/java/net/md_5/bungee/connection/InitialHandler.java
@@ -274,7 +274,6 @@ public class InitialHandler extends PacketHandler implements PendingConnection
         }
 
         this.virtualHost = InetSocketAddress.createUnresolved( handshake.getHost(), handshake.getPort() );
-        bungee.getLogger().log( Level.INFO, "{0} has connected", this );
 
         bungee.getPluginManager().callEvent( new PlayerHandshakeEvent( InitialHandler.this, handshake ) );
 
@@ -282,10 +281,15 @@ public class InitialHandler extends PacketHandler implements PendingConnection
         {
             case 1:
                 // Ping
+                if ( BungeeCord.getInstance().getConfig().isLogServerListPing() )
+                {
+                    bungee.getLogger().log( Level.INFO, "{0} has pinged", this );
+                }
                 thisState = State.STATUS;
                 ch.setProtocol( Protocol.STATUS );
                 break;
             case 2:
+                bungee.getLogger().log( Level.INFO, "{0} has connected", this );
                 thisState = State.USERNAME;
                 ch.setProtocol( Protocol.LOGIN );
                 // Login
@@ -618,6 +622,12 @@ public class InitialHandler extends PacketHandler implements PendingConnection
     @Override
     public String toString()
     {
-        return "[" + ( ( getName() != null ) ? getName() : getAddress() ) + "] <-> InitialHandler";
+        if ( getName() != null )
+        {
+            return "[" + getAddress() + "|" + getName() + "] <-> InitialHandler";
+        } else
+        {
+            return "[" + getAddress() + "] <-> InitialHandler";
+        }
     }
 }
diff --git a/proxy/src/main/java/net/md_5/bungee/connection/UpstreamBridge.java b/proxy/src/main/java/net/md_5/bungee/connection/UpstreamBridge.java
index e82f2d2..7b7d729 100644
--- a/proxy/src/main/java/net/md_5/bungee/connection/UpstreamBridge.java
+++ b/proxy/src/main/java/net/md_5/bungee/connection/UpstreamBridge.java
@@ -207,6 +207,6 @@ public class UpstreamBridge extends PacketHandler
     @Override
     public String toString()
     {
-        return "[" + con.getName() + "] -> UpstreamBridge";
+        return "[" + con.getAddress() + "|" + con.getName() + "] -> UpstreamBridge";
     }
 }
-- 
1.9.5.msysgit.0

